notification-server/src/bin/service.rs

```diff
@@ -87,9 +93,11 @@ async fn traverse_chain(
     database_connection: &DatabaseConnection,
     concordium_client: &mut Client,
     gcloud: &GoogleCloud<CustomServiceAccount>,
-    mut receiver: impl Stream<Item = Result<FinalizedBlockInfo, tonic::Status>> + Unpin,
-) {
-    while let Some(v) = receiver.next().await {
+    mut receiver: FinalizedBlocksStream,
+    process_timeout: Duration,
+    mut height: AbsoluteBlockHeight,
+) -> AbsoluteBlockHeight {
```

Comment:
Cannot comment on the complete fn signature, but I think a doc comment would be good to explain the different scenarios such as
- in which case it returns/crashes
- when transactions are skipped, i.e. any error when handling specific transactions (is this even a good approach to handling these errors?)

Reply:
Very nice!